vendor/bat-native-confirmations/src/bat/confirmations/internal/redeem_token.cc

```diff
@@ -301,7 +292,10 @@ void RedeemToken::OnFetchPaymentToken(
   base::DictionaryValue* payment_token_dictionary;
   if (!payment_token_value->GetAsDictionary(&payment_token_dictionary)) {
     BLOG(ERROR) << "Response missing paymentToken dictionary";
-    OnRedeem(FAILED, confirmation);
+    OnRedeem(FAILED, confirmation, false);
+
+    // Token is in a bad state so redeem a new token
+    Redeem(confirmation.creative_instance_id, confirmation.type);
```

Comment:
I'd say the comment is superfluous as we know we're handling a fail case.

Reply:
We don't always redeem a new token when in a fail state, so wanted to make clear why we are redeeming a new token here